src/components/app/data/hooks/useSearchCatalogs.test.js

```diff
@@ -37,34 +44,50 @@ jest.mock('../../../../config', () => ({
   features: { ENROLL_WITH_CODES: true },
 }));
 
-const mockEnterpriseCustomer = enterpriseCustomerFactory();
 const mockSubscriptionCatalog = 'test-subscription-catalog-uuid';
 const mockCouponCodeCatalog = 'test-coupon-code-catalog-uuid';
 const mockEnterpriseOfferCatalog = 'test-enterprise-offer-catalog-uuid';
 const mockPolicyCatalog = 'test-policy-catalog-uuid';
+const defaultAppContextValue = {
+  authenticatedUser: authenticatedUserFactory(),
+};
 
 describe('useSearchCatalogs', () => {
+  const Wrapper = ({ children }) => (
+    // eslint-disable-next-line react/jsx-filename-extension
+    <AppContext.Provider value={defaultAppContextValue}>
+      {children}
+    </AppContext.Provider>
+  );
   beforeEach(() => {
     jest.clearAllMocks();
     useEnterpriseCustomer.mockReturnValue({ data: mockEnterpriseCustomer });
     useRedeemablePolicies.mockReturnValue({ data: { redeemablePolicies: [] } });
-    useCatalogsForSubsidyRequests.mockReturnValue({ catalogsForSubsidyRequests: [] });
+    useCatalogsForSubsidyRequests.mockReturnValue([]);
     useEnterpriseOffers.mockReturnValue({ data: { currentEnterpriseOffers: [] } });
     useSubscriptions.mockReturnValue({ data: { subscriptionLicense: [] } });
     useCouponCodes.mockReturnValue({ data: { couponCodeAssignments: [] } });
   });
 
-  it.each('should include catalog from subscription (%s)', ({ isSubscriptionPlanExpired }) => {
+  it.each([
+    {
+      isSubscriptionPlanExpired: true,
+    },
+    {
+      isSubscriptionPlanExpired: false,
+    },
+  ])('should include catalog from subscription (%s)', ({ isSubscriptionPlanExpired }) => {
     const mockSubscriptionLicense = {
-      status: LICENSE_STATUS.ACTIVATED,
+      status: isSubscriptionPlanExpired ? LICENSE_STATUS.ASSIGNED : LICENSE_STATUS.ACTIVATED,
```

Comment:
[clarification] A learner still might have an activated license with an expired license. Just curious about the conditional change in `status` based on `isSubscriptionPlanExpired`?

Reply:
As part of resolving this issue, it was noted that the `useSubscriptions` returns subscription licenses and plans that could be expired. This resulted of the side effect where a expired but active subscription license could be added to the `useSearchCatalogs` array return.

The following modification were made to `getSearchCatalogs` function to account for expired subscription licenses.

```js
if (subscriptionLicense?.subscriptionPlan.isCurrent && subscriptionLicense?.status === LICENSE_STATUS.ACTIVATED) {
    catalogUUIDs.add(subscriptionLicense.subscriptionPlan.enterpriseCatalogUuid);
  }
```